urldeb33fd2fa_friedemann_distripedia_tgz-pJ8-org_json_TestJ8/src/grinder/controller/ControllerGUI.java

```diff
@@ -0,0 +1,763 @@
+
+		table_clients = new JTable(ctModel);
+
+		// set column widths
+		final TableColumn column = table_clients.getColumnModel().getColumn(1);
+		column.setPreferredWidth(220);
+
+		scroll_tablePane = new JScrollPane(table_clients);
+		// table_clients.setFillsViewportHeight(true);
+
+		panel_clients.add(scroll_tablePane);
+
+		panel_clients.setMinimumSize(new Dimension(0, 220));
+		setTitledBorder(panel_clients, "connected clients");
+	}
+
+	/**
+	 * Builds the test settings panel.
+	 */
+	private void buildTestSettingsPanel() {
+		panel_testSetup = new JPanel(new GridLayout(4, 2));
+
+		slider_userHTTP = createTitledSilder(ControllerGUI.SLIDER_TITLE_USER_HTTP + ": 0", 0,
+				16000, 0);
+		slider_userWebSocket = createTitledSilder(ControllerGUI.SLIDER_TITLE_USER_WSOCKET
+				+ ": 0", 0, 16000, 0);
+		slider_userStratus = createTitledSilder(ControllerGUI.SLIDER_TITLE_USER_STRATUS + ": 0",
+				0, 16000, 0);
+		slider_minTabOpenTime = createTitledSilder(ControllerGUI.SLIDER_TITLE_TAB_MIN + ": 0",
+				0, 3600, 0);
+		slider_maxTabOpenTime = createTitledSilder(ControllerGUI.SLIDER_TITLE_TAB_MAX + ": 0",
+				0, 3600, 0);
+		slider_minReadingTime = createTitledSilder(ControllerGUI.SLIDER_TITLE_READ_MIN + ": 0",
+				0, 3600, 0);
+		slider_maxReadingTime = createTitledSilder(ControllerGUI.SLIDER_TITLE_READ_MAX + ": 0",
+				0, 3600, 0);
+		slider_testDuration = createTitledSilder(ControllerGUI.SLIDER_TITLE_TEST_DUR + ": 0", 0,
+				3600, 0);
+
+		panel_testSetup.add(slider_userHTTP);
+		panel_testSetup.add(slider_minTabOpenTime);
+
+		panel_testSetup.add(slider_userWebSocket);
+		panel_testSetup.add(slider_maxTabOpenTime);
+
+		panel_testSetup.add(slider_userStratus);
+		panel_testSetup.add(slider_minReadingTime);
+
+		panel_testSetup.add(slider_testDuration);
+		panel_testSetup.add(slider_maxReadingTime);
+
+		panel_testSetup.setMinimumSize(new Dimension(0, 230));
+		setTitledBorder(panel_testSetup, "test case setup");
+	}
+
+	/**
+	 * Builds the control panel.
+	 */
+	private void buildControlPanel() {
+		panel_controls = new JPanel(new GridBagLayout());
+
+		// article list
+		final JLabel lbl_articleList = new JLabel("Article list status: ");
+		lbl_articleListStatus = new JLabel("not loaded", SwingConstants.CENTER);
+		lbl_articleListStatus.setFont(new Font("Arial", Font.BOLD, 12));
+		final JButton btn_articleList = new JButton(action_loadArticleList);
+
+		// web server address
+		final JLabel lbl_webServerAddress = new JLabel("Webserver address: ");
+		field_webServerAddress = new JTextField();
+
+		// web server address
+		final JLabel lbl_webSocketAddress = new JLabel("Websocket service address: ");
+		field_webSocketURI = new JTextField();
+
+		// test id
+		final JLabel lbl_testID = new JLabel("Test id: ");
+		field_testID = new JTextField("run01");
+
+		// start time
+		final JLabel lbl_startTime = new JLabel("Test start time: ");
+		final Format f = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.